python/prairielearn_test.py

```diff
@@ -88,26 +89,82 @@ def test_set_score_data(
     assert math.isclose(question_data["score"], expected_score)
 
 
-class TestEnum(Enum):
+@pytest.mark.parametrize(
+    "numpy_object",
+    [
+        np.int64(5),
+        np.int32(-12),
+        np.uint8(55),
+        np.byte(3),
+        np.float128(-1100204.04010340),
+        np.float32(2.1100044587483),
+        np.float16(0.00000184388328),
+        np.int64((2**53) + 5),
+        np.complex128("12+3j"),
+        np.complex256("12+3j"),
+        np.arange(15),
+        np.array([1.2, 3.5, 5.1]),
+        np.array([1, 2, 3, 4]),
+        np.array([(1.5, 2, 3), (4, 5, 6)]),
+        np.array([[1, 2], [3, 4]], dtype=complex),
+        np.array([[1, "stuff"], [3, None]], dtype=object),
+        np.ones((2, 3, 4), dtype=np.int16),
+    ],
+)
+def test_numpy_serialization(numpy_object: Any) -> None:
+    """Test equality after conversion of various numpy objects."""
+
+    json_object = json.dumps(
+        pl.to_json(numpy_object, np_encoding_version=2), allow_nan=False
+    )
+    decoded_json_object = pl.from_json(json.loads(json_object))
+
+    assert type(numpy_object) == type(decoded_json_object)
+    np.testing.assert_array_equal(numpy_object, decoded_json_object, strict=True)
+
+
+@pytest.mark.parametrize(
+    "object_to_encode, expected_result",
+    [(np.float64(5.0), 5.0), (np.complex128("12+3j"), complex("12+3j"))],
+)
+def test_legacy_serialization(object_to_encode: Any, expected_result: Any) -> None:
+    """Test that nothing happens under the old encoding for numpy scalars."""
+
+    json_object = json.dumps(pl.to_json(object_to_encode), allow_nan=False)
+    decoded_json_object = pl.from_json(json.loads(json_object))
+
+    assert decoded_json_object == expected_result
+
+
+class DummyEnum(Enum):
```

Comment:
Had to change this because pytest was trying to pick this up as a test class before, whoops.